tablenamer/plural/enplural.go

```diff
@@ -0,0 +1,198 @@
+package plural
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+)
+
+// regCache holds a compiled regex to match pluralized word
+type regCache struct {
+	Regexp   *regexp.Regexp
+	Replacer string
+}
+
+// irregular holds irregular words
+type irregular struct {
+	Singular string
+	Plural   string
+}
+
+// EnglishPlural defines methods and rules for getting plural form of a word for English
+type EnglishPlural struct {
```

Comment:
The cache will be more efficient with https://golang.org/pkg/sync/#Map
(see another similar comment in this review)

I wonder about the need to lock things while adding / accessing rules, I think it's better to do it one **before** computing any table name. I'll explain it in a comment into the PR.